src/compiler/checker.ts

```diff
@@ -27730,6 +27755,28 @@ namespace ts {
             }
         }
 
+        function checkPrivateIdentifierExpression(privId: PrivateIdentifier): Type {

```

Comment:
I'd rename this `checkGrammarPrivateIdentifierExpression` and have it return boolean. Then invert the `ifs` and `return grammarErrorOnNode(...`, which returns true if the error could be added. This will make it look like the other checkGrammar* functions, which it is already most similar to.


The symbol-setting code should be in a separate function that's called in `getSymbolOfNameOrPropertyAccessExpression` and `narrowTypeByPrivateIdentifierInInExpression`, while `checkGrammarPrivateIdentifierExpression` should be called from `checkExpressionWorker` (and it can just `return anyType` from inside the switch).

Reply:
Done. One difference was that I still kept a `checkPrivateIdentifierExpression` which keeps the `markPropertyAsReferenced` and is only called from `checkExpressionWorker`. As it started to do more than just call `checkGrammarPrivateIdentifierExpression` I didn't inline it into the switch. But happy to if you'd prefer?